webclient/app/src/app/features/apps/entities/entityDiagram/EntityDiagram.jsx

```diff
@@ -64,35 +77,67 @@ function EntityDiagram(props) {
         return entityRest.updatePositionByApp(appId, entity);
     }
 
+    function deleteEnum(enumId) {
+        if (!editable) {
+            return;
+        }
+        return enumRest.delete(enumId)
+            .then(reloadEnums);
+    }
+
+    function updateEnum(enumDef) {
+        if (!editable) {
+            return;
+        }
+        return enumRest.createByApp(appId, enumDef).then(reloadEnums);
+    }
+
     function reloadEntities() {
         if (!editable) {
             return;
         }
         return entityRest.findAllEntitiesByApp(appId).then(response => {
             const newEntities = response.data;
-            onChange(newEntities);
+            onChangeEntities(newEntities);
             return newEntities;
         });
     }
 
+    function reloadEnums() {
+        if (!editable) {
+            return;
+        }
+        return enumRest.findAllByApp(appId).then(response => {
+            const newEnums = response.data;
+            onChangeEnums(newEnums);
+            return newEnums;
+        });
+    }
+
     function prepareUpdateEntity(updatedEntity) {
         if (!editable) {
             return;
         }
-        const createEntity = updateEntity(updatedEntity).then(reloadEntities);
-        return createEntity;
+        return updateEntity(updatedEntity).then(reloadEntities);
     }
 
     function savePosition(entity, index, draggableData) {
         const updatedEntity = updatePosition(entity, draggableData);
         const newEntities = [...entities];
         newEntities[index] = updatedEntity;
-        onChange(newEntities);
+        onChangeEntities(newEntities);
         updatePositionInDB(updatedEntity);
     }
 
+    function saveEnumPosition(enums, enumDef, index, draggableData) {
+        const enumToUpdate = updateEnumPosition(enumDef, draggableData);
+        const enumList = updateEnumInList(enums, enumToUpdate, index);
```

Comment:
Wahrscheinlich ist dieser Kommentar irrelevant, aber wofür braucht man noch den index? Hat nicht jedes Enum eine id direkt integriert?

Reply:
um das geänderte Enum bereits vor dem Speichern in der liste zu aktualisieren

Reply:
Könntest du auch mit find() und der id machen...